lib/gateway/Shard.js

```diff
@@ -669,12 +669,13 @@ class Shard extends EventEmitter {
                 if(packet.d.user.username !== undefined) {
                     let user = this.client.users.get(packet.d.user.id);
                     let oldUser = null;
-                    if(user && (user.username !== packet.d.user.username || user.discriminator !== packet.d.user.discriminator || user.avatar !== packet.d.user.avatar || user.globalName !== packet.d.user.global_name)) {
+                    if(user && (user.username !== packet.d.user.username || user.discriminator !== packet.d.user.discriminator || user.avatar !== packet.d.user.avatar || user.globalName !== packet.d.user.global_name || user.avatarDecoration !== packet.d.user.avatar_decoration)) {
```

Comment:
Probably forgot to update that here

```suggestion
                    if(user && (user.username !== packet.d.user.username
                        || user.discriminator !== packet.d.user.discriminator
                        || user.avatar !== packet.d.user.avatar
                        || user.globalName !== packet.d.user.global_name
                        || user.avatarDecorationData?.asset !== packet.d.user.avatar_decoration_data?.asset
                    )) {
```